investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java

```diff
@@ -402,36 +403,41 @@ sending reporting(nrt) beans as json to kafka
-        investigationInterviewReporting.setRecordStatusCd(interview.getRecordStatusCd());
-        investigationInterviewReporting.setLocalId(interview.getLocalId());
-        investigationInterviewReporting.setVersionCtrlNbr(interview.getVersionCtrlNbr());
-        investigationInterviewReporting.setInvestigationUid(interview.getInvestigationUid());
-        investigationInterviewReporting.setOrganizationUid(interview.getOrganizationUid());
-        investigationInterviewReporting.setProviderUid(interview.getProviderUid());
-        investigationInterviewReporting.setPatientUid(interview.getPatientUid());
-        return investigationInterviewReporting;
+        InterviewReporting investigationReporting = new InterviewReporting();
+        investigationReporting.setInterviewUid(interview.getInterviewUid());
+        investigationReporting.setInterviewDate(interview.getInterviewDate());
+        investigationReporting.setInterviewLocCd(interview.getInterviewLocCd());
+        investigationReporting.setInterviewTypeCd(interview.getInterviewTypeCd());
+        investigationReporting.setInterviewStatusCd(interview.getInterviewStatusCd());
+        investigationReporting.setIntervieweeRoleCd(interview.getIntervieweeRoleCd());
+        investigationReporting.setIxIntervieweeRole(interview.getIxIntervieweeRole());
+        investigationReporting.setAddTime(interview.getAddTime());
+        investigationReporting.setAddUserId(interview.getAddUserId());
+        investigationReporting.setIxLocation(interview.getIxLocation());
+        investigationReporting.setIxStatus(interview.getIxStatus());
+        investigationReporting.setIxType(interview.getIxType());
+        investigationReporting.setLastChgTime(interview.getLastChgTime());
+        investigationReporting.setLastChgUserId(interview.getLastChgUserId());
+        investigationReporting.setRecordStatusTime(interview.getRecordStatusTime());
+        investigationReporting.setRecordStatusCd(interview.getRecordStatusCd());
+        investigationReporting.setLocalId(interview.getLocalId());
+        investigationReporting.setVersionCtrlNbr(interview.getVersionCtrlNbr());
+        investigationReporting.setInvestigationUid(interview.getInvestigationUid());
+        investigationReporting.setOrganizationUid(interview.getOrganizationUid());
+        investigationReporting.setProviderUid(interview.getProviderUid());
+        investigationReporting.setPatientUid(interview.getPatientUid());
+        return investigationReporting;
     }
 
     public void transformAndSendInterviewAnswer(Interview interview) {
         try {
             JsonNode answerArray = parseJsonArray(interview.getAnswers());
 
+            // Tombstone message to delete all interview answers for specified interview uid
+            String jsonKey1 = jsonGenerator.generateStringJson(new InterviewReportingKey(interview.getInterviewUid()));
+            kafkaTemplate.send(interviewAnswerOutputTopicName, jsonKey1, null);
```

Comment:
Tombstone message should be sent _before_ `parseJsonArray` (it will throw exception, if all answers are deleted)

Reply:
this is actually a very good find. I have made the changes.